code/modules/psionics/psion.dm

```diff
@@ -78,22 +82,68 @@
 		if(owner.stats.getPerk(PERK_PSI_PSYCHOLOGIST))
 			psi_max_bonus += 5
 
+		psi_max_bonus += psionic_equipment_check("psi_max_bonus")
+
 		max_psi_points = round(clamp((owner.stats.getStat(STAT_COG) * 0.1), 1, 30)) + psi_max_bonus
 
-		cognitive_potential = round(clamp((owner.stats.getStat(STAT_COG) * 0.5), 0, 3))
+		cognitive_potential = round(clamp((owner.stats.getStat(STAT_COG) * 0.1), 0, cognitive_potential_max), 0.1)
 
+		var/regen_points_timer = (5 MINUTES - cognitive_potential MINUTES)
+		var/min_timer = 1 MINUTES
 
 		if(owner.stats.getPerk(PERK_PSI_GRACE))
-			addtimer(CALLBACK(src, PROC_REF(regen_points)), (5 MINUTES - cognitive_potential MINUTES) * 0.5)
-		else
-			addtimer(CALLBACK(src, PROC_REF(regen_points)), (5 MINUTES - cognitive_potential MINUTES))
+			regen_points_timer *= 0.5
+
+		regen_points_timer -= (psionic_equipment_check("psi_regen_helpers") SECONDS)
+
+		regen_points_timer -= cognitive_potential MINUTES
+
+		if(min_timer > regen_points_timer)
+			regen_points_timer = min_timer
+
+		addtimer(CALLBACK(src, PROC_REF(regen_points)), regen_points_timer)
 
 		if(psi_points < max_psi_points)
 			psi_points += 1
 
 		if(owner.psi_blocking < 0) //resets psiblock to zero so people can't somehow farm it into the negatives.
 			owner.psi_blocking = 0
 
+//The major types are:
+//"psi_max_bonus"
+//"psi_regen_helpers" - This is in seconds!
+//"cognitive_potential_max_bonus" - caps at 4
+
+/obj/item/organ/internal/psionic_tumor/proc/psionic_equipment_check(type)
```

Comment:
This would be better written if the above types were made into variables owned by the tumor, that instead are updated by this function. The updated variables can then later be used to calculate the bobuses. That way, it is only called once per regen points call.

Reply:
I want to keep the framework for it like this rather then bloat the parent